lms/services/lti_h.py

```diff
@@ -118,6 +96,6 @@ def _upsert_h_group(self, group_id, group_name, creator):
                 raise HTTPBadRequest("Instructor must launch assignment first.")
 
         # Try to create the group with the current instructor as its creator.
-        self.h_api.create_group(
-            group_id=group_id, group_name=group_name, creator=creator
+        self.h_api.upsert_group(
+            group_id=group.groupid, group_name=group.name, creator=creator
```

Comment:
There's something super fishy going on with the logic in this function. It's more obvious with the rename to `upsert_group`. I think we are quietly always an instructor by this point, which means we probably try a pointless "create" for instructors above.

If we flipped this around to test whether you were an instructor first, we could probably skip that. This doesn't matter if we end up deleting the lot though.